//##############################################################################
//#  Copyright (c) 2019 Intel Corporation
//# 
//# Licensed under the Apache License, Version 2.0 (the "License");
//#  you may not use this file except in compliance with the License.
//#  You may obtain a copy of the License at
//# 
//#      http://www.apache.org/licenses/LICENSE-2.0
//# 
//#  Unless required by applicable law or agreed to in writing, software
//#  distributed under the License is distributed on an "AS IS" BASIS,
//#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//#  See the License for the specific language governing permissions and
//#  limitations under the License.
//##############################################################################
//#    File Abstract: 
//#    This is the main/entry point file for the Minion data collector program
//#
//##############################################################################
syntax = "proto3";

package intel.node.frequency;

// Interface exported by our 'Services'
service NodeFrequencyManagerService
{
    rpc Set_Core_Frequency(SetFrequencyRequest) returns (ServiceResponse) {};
    rpc Set_AllCore_Frequency(SetFrequencyRequest) returns (ServiceResponse) {};
    rpc Set_Core_Percent_Frequency(SetFrequencyPercentRequest) returns (ServiceResponse) {};
    rpc Set_All_CorePercent_Frequency(SetFrequencyPercentRequest) returns (ServiceResponse) {};
    rpc Get_Core_Frequency_Info(CoreNumber) returns (CoreFrequencyInfo) {};
    rpc Set_Random_Frequencies(Empty) returns (ServiceResponse) {};
    rpc Set_SineWave_Frequencies(Empty) returns (ServiceResponse) {};
}

message SetFrequencyRequest
{
    int32 CoreNumber = 1;
    int32 Frequency = 2;
}

message CoreNumber
{
    int32 CoreNumber = 1;
}

message SetFrequencyPercentRequest
{
    int32 CoreNumber = 1;
    float Frequency = 2;
}

message CoreFrequencyInfo
{
    ServiceResponse Response = 1;
    int32 CoreNumber = 2;
    int32 MaxFrequency = 3;
    int32 MinFrequency = 4;
    int32 CurrentFrequency = 5;
}

message ServiceResponse
{
    bool Success = 1;
    string Reason = 2;
}

message Empty
{
}